### PR TITLE
add option to disable headercheck and add  in /health

### LINKED ISSUE #2095: Option to disable the Host / Origin / Referer checks

### PR DESCRIPTION
**Pull request by therealssj**
Fixes #2095

Changes:
- Add option `-disable-header-check` to disable header check for host/origin/referer header and add `header_check_enabled` in `/health`

Does this change need to mentioned in CHANGELOG.md?
Yes

### CODE REVIEW

**Comment by therealssj** [comment]
Should I add integration tests with cors disabled just like csrf? 🤔

**Comment by gz-c** [comment]
yes but the tests should care more that cors works enabled

**Comment by gz-c** [comment]
This does not fix #2095, which refers to the Host/Origin/Refer checks, not CORS, although it is related.

CORS cannot be disabled; if no CORS headers are specified, the browser defaults to a default CORS policy.

**Comment by gz-c** [comment]
Update the initial PR comment text

File: src/skycoin/config.go [hunk 1]
```diff
@@ -587,6 +591,7 @@ func (c *NodeConfig) RegisterFlags() {
 	flag.BoolVar(&c.EnableGUI, "enable-gui", c.EnableGUI, "Enable GUI")
 	flag.BoolVar(&c.EnableUnversionedAPI, "enable-unversioned-api", c.EnableUnversionedAPI, "Enable the deprecated unversioned API endpoints without /api/v1 prefix")
 	flag.BoolVar(&c.DisableCSRF, "disable-csrf", c.DisableCSRF, "disable CSRF check")
+	flag.BoolVar(&c.DisableHeadercheck, "disable-headercheck", c.DisableHeadercheck, "disable Header check")
```
Comment (gz-c) [inline src/skycoin/config.go]: Needs a better description, "Header check" is ambiguous, it disabled the Host, Origin and Referer checks

File: CHANGELOG.md [hunk 2]
```diff
@@ -34,6 +34,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
   Now all options of a cli command must only use `--` prefix instead of a mix of `--` and `-` prefixes.
   `-` prefix is only allowed when using shorthand notation.
 - Use an optimized `base58` library for faster address decoding and encoding.
+- Add option to disable host/origin/referer header checks and add `headercheck_enabled` in `/health`.
```
Comment (gz-c) [inline CHANGELOG.md]: Mention the option name and usage here

File: src/api/http.go [hunk 3]
```diff
@@ -68,18 +68,19 @@ type Server struct {
...
-	WriteTimeout   time.Duration
-	IdleTimeout    time.Duration
-	Health         HealthConfig
-	HostWhitelist  []string
-	EnabledAPISets map[string]struct{}
-	Username       string
-	Password       string
+	StaticDir          string
+	DisableCSRF        bool
+	DisableHeadercheck bool
```
Comment (gz-c) [inline src/api/http.go]: `DisableHeaderCheck`

File: src/api/health.go [hunk 4]
```diff
@@ -76,6 +77,7 @@ func healthHandler(c muxConfig, gateway Gatewayer) http.HandlerFunc {
 			IncomingConnections:  health.IncomingConnections,
 			Uptime:               wh.FromDuration(health.Uptime),
 			CSRFEnabled:          !c.disableCSRF,
+			HeadercheckEnabled:   !c.disableHeadercheck,
```
Comment (gz-c) [inline src/api/health.go]: `disabledHeaderCheck`

File: src/api/http.go [hunk 5]
```diff
@@ -349,20 +356,26 @@ func newServerMux(c muxConfig, gateway Gatewayer) *http.ServeMux {
 		}
 	}
 
-	webHandlerCSRFOptional := func(apiVersion, endpoint string, handler http.Handler, checkCSRF bool) {
+	webHandlerWithOptionals := func(apiVersion, endpoint string, handler http.Handler, checkCSRF bool, checkHeaders bool) {
```
Comment (gz-c) [inline src/api/http.go]: can remove the first `bool`

File: src/api/http_test.go [hunk 6]
```diff
@@ -26,11 +26,12 @@ var allAPISetsEnabled = map[string]struct{}{
...
 	return muxConfig{
-		host:           configuredHost,
-		appLoc:         ".",
-		disableCSRF:    true,
-		disableCSP:     true,
-		enabledAPISets: allAPISetsEnabled,
+		host:               configuredHost,
+		appLoc:             ".",
+		disableCSRF:        true,
+		disableHeadercheck: true,
```
Comment (gz-c) [inline src/api/http_test.go]: Is it necessary to have it true by default? Don't the tests already cope with the header check

File: src/skycoin/config.go [hunk 7]
```diff
@@ -54,6 +54,10 @@ type NodeConfig struct {
 	EnableGUI bool
 	// Disable CSRF check in the wallet API
 	DisableCSRF bool
+	// Disable Host, Origin and Referer header check in the wallet API
+	DisableHeadercheck bool
+	// Enable unversioned API endpoints (without the /api/v1 prefix)
```
Comment (gz-c) [inline src/skycoin/config.go]: EnableUnversionedAPI was removed

File: src/skycoin/config.go [hunk 8]
```diff
@@ -54,6 +54,10 @@ type NodeConfig struct {
 	EnableGUI bool
 	// Disable CSRF check in the wallet API
 	DisableCSRF bool
+	// Disable Host, Origin and Referer header check in the wallet API
+	DisableHeadercheck bool
```
Comment (gz-c) [inline src/skycoin/config.go]: `DisableHeaderCheck`

**Pull request merged**